### PR TITLE
feature/MANG-253-move-extended-url-rule-to-externals

### PR DESCRIPTION
**Pull request by egor-dev**
- Add "extended_url" rule to the Validator

### CODE REVIEW

File: src/Bridge/Laravel/Validation/ExtendedUrlRule.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
...jabber|jar|jms|keyparc|lastfm|ldap|ldaps|magnet|mailserver|mailto|maps|market|message|mid|mms|modem|ms-help|ms-settings|ms-settings-airplanemode|ms-settings-bluetooth|ms-settings-camera|ms-settings-cellular|ms-settings-cloudstorage|ms-settings-emailandaccounts|ms-settings-language|ms-settings-location|ms-settings-lock|ms-settings-nfctransactions|ms-settings-notifications|ms-settings-power|ms-settings-privacy|ms-settings-proximity|ms-settings-screenrotation|ms-settings-wifi|ms-settings-workplace|msnim|msrp|msrps|mtqp|mumble|mupdate|mvn|news|nfs|ni|nih|nntp|notes|oid|opaquelocktoken|pack|palm|paparazzi|pkcs11|platform|pop|pres|prospero|proxy|psyc|query|redis|rediss|reload|res|resource|rmi|rsync|rtmfp|rtmp|rtsp|rtsps|rtspu|secondlife|s3|service|session|sftp|sgn|shttp|sieve|sip|sips|skype|smb|sms|smtp|snews|snmp|soap.beep|soap.beeps|soldat|spotify|ssh|steam|stun|stuns|submit|svn|tag|teamspeak|tel|teliaeid|telnet|tftp|things|thismessage|tip|tn3270|turn|turns|tv|udp|unreal|urn|ut2004|vemmi|ventrilo|videotex|view-source|wais|webcal|ws|wss|wtai|wyciwyg|xcon|xcon-userid|xfire|xmlrpc\.beep|xmlrpc.beeps|xmpp|xri|ymsgr|z39\.50|z39\.50r|z39\.50s)):)?//                                 # protocol
```
Comment (merk) [inline src/Bridge/Laravel/Validation/ExtendedUrlRule.php]: I dont agree with the protocol options being listed like this: there can be all kinds of custom protocols and we should be using a simpler constraint:

https://github.com/symfony/validator/blob/master/Constraints/UrlValidator.php

File: tests/Bridge/Laravel/Validation/ExtendedUrlRuleTest.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * @covers \EoneoPay\Externals\Bridge\Laravel\Validation\ExtendedUrlRule
+ */
+final class ExtendedUrlRuleTest extends TestCase
+{
+    /**
+     * Data provider for testing extended_url rule.
+     *
+     * @return mixed[]
+     */
+    public function getValidationData(): iterable
```
Comment (merk) [inline tests/Bridge/Laravel/Validation/ExtendedUrlRuleTest.php]: I'd like to see a more exhaustive test of invalid urls:

https://github.com/symfony/validator/blob/master/Tests/Constraints/UrlValidatorTest.php#L215-L259

File: src/Bridge/Laravel/Validation/ExtendedUrlRule.php (hunk 1 above)
Comment (egor-dev) [reply src/Bridge/Laravel/Validation/ExtendedUrlRule.php]: In your example pattern will eventually contain only http/https after sprintf. Do you want me to put only http/https there? Or allow any alphanum string (with dashes and dots)?

Comment (egor-dev) [reply src/Bridge/Laravel/Validation/ExtendedUrlRule.php]: @merk

**Comment by merk** [comment]
@egor-dev I dont see any additional commits addressing my comments? Additionally, the build has failed and the branch is out of date.

Can you please rebase this PR on the v1.x branch and change its target to v1.x

**Pull request closed without merging**